frontend/src/scenes/data-warehouse/new/sourceWizardLogic.tsx

```diff
@@ -433,5 +441,105 @@ export const sourceWizardLogic = kea<sourceWizardLogicType>([
+        handleRedirect: async ({ kind, searchParams }) => {
+            switch (kind) {
+                case 'hubspot': {
+                    actions.updateSource({
+                        source_type: 'Hubspot',
+                        payload: {
+                            code: searchParams.code,
+                            redirect_uri: getHubspotRedirectUri(),
+                        },
+                    })
+                    return
+                }
+                default:
+                    lemonToast.error(`Something went wrong.`)
+            }
+        },
+        submitSourceConnectionDetailsSuccess: () => {
+            actions.getDatabaseSchemas()
+        },
+        getDatabaseSchemas: async () => {
+            if (values.selectedConnector) {
+                const schemas = await api.externalDataSources.database_schema(
+                    values.selectedConnector.name,
+                    values.source.payload ?? {}
+                )
+                actions.setDatabaseSchemas(schemas)
+                actions.onNext()
+            }
+        },
+    })),
+    urlToAction(({ actions }) => ({
+        '/data-warehouse/:kind/redirect': ({ kind = '' }, searchParams) => {
+            if (kind === 'hubspot') {
+                router.actions.push(urls.dataWarehouseTable(), { kind, code: searchParams.code })
+            }
+        },
+        '/data-warehouse/new': (_, searchParams) => {
+            if (searchParams.kind == 'hubspot' && searchParams.code) {
+                actions.selectConnector(SOURCE_DETAILS['Hubspot'])
+                actions.handleRedirect(searchParams.kind, {
+                    code: searchParams.code,
+                })
+                actions.setStep(2)
+            }
+        },
+    })),
+    forms(({ actions, values }) => ({
+        sourceConnectionDetails: {
+            defaults: {
+                prefix: '',
+                payload: {},
+            } as Partial<ExternalDataSourceCreatePayload>,
+            submit: async (sourceValues) => {
+                if (values.selectedConnector) {
+                    const payload = {
+                        ...sourceValues,
+                        source_type: values.selectedConnector.name,
+                    }
+                    actions.setIsLoading(true)
+
+                    try {
+                        await api.externalDataSources.source_prefix(payload.source_type, payload.prefix)
+                        actions.updateSource(payload)
+                        actions.setIsLoading(false)
+                    } catch (e: any) {
+                        if (e?.data?.message) {
+                            actions.setSourceConnectionDetailsManualErrors({ prefix: e.data.message })
+                        }
+                        actions.setIsLoading(false)
+
+                        throw e
+                    }
+                }
+            },
+        },
     })),
 ])
+
+// const getErrorsForFields = (
```

Comment:
didn't manage to get this back in the form errors above. somehow any values that i try to access in defaults/errors causes the logic to not mount/connect

Reply:
Managed to get around this, no idea why it wasn't working - makes me think we're mounting the logic wrong, but everything looked good, so, 🤷